app/models/tag.rb

```diff
@@ -101,14 +101,9 @@ def self.find_nodes_by_type(tagnames, type = 'note', limit = 10)
     # .select(%i[node.nid node.status node.type community_tags.nid community_tags.tid term_data.name term_data.tid])
     # above select could be added later for further optimization
     # .where('term_data.name IN (?) OR term_data.parent in (?)', tagnames, tagnames) # greedily fetch children
-    tags = Tag.where('term_data.name IN (?)', tagnames)
-    parents = Node.where(status: 1, type: type)
-                  .includes(:tag)
-                  .references(:term_data)
-                  .where('term_data.name IN (?)', tags.collect(&:parent))
     order = 'node_revisions.timestamp DESC'
     order = 'created DESC' if type == 'note'
-    Node.where('node.nid IN (?)', (nodes + parents).collect(&:nid))
+    Node.where('node.nid IN (?)', nodes.collect(&:nid))
```

Comment:
Nicely done here! Actually we should be able to remove the comment on line 103 too, also related to parenting:

>     # .where('term_data.name IN (?) OR term_data.parent in (?)', tagnames, tagnames) # greedily fetch children